receiver/couchdbreceiver/client.go

```diff
@@ -0,0 +1,125 @@
+package couchdbreceiver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+// CouchDBClient defines the basic HTTP client interface.
+type CouchDBClient interface {
+	BuildReq(path string) (*http.Request, error)
+	Get(url string) ([]byte, error)
+	GetNodeNames() ([]string, error)
+	GetStats(nodeName string) (map[string]interface{}, error)
+}
+
+var _ CouchDBClient = (*couchDBClient)(nil)
+
+type couchDBClient struct {
+	client *http.Client
+	cfg    *Config
+	logger *zap.Logger
+}
+
+// NewCouchDBClient creates a new client to make requests for the CouchDB receiver.
+func NewCouchDBClient(cfg *Config, host component.Host, logger *zap.Logger) (CouchDBClient, error) {
+	client, err := cfg.ToClient(host.GetExtensions())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP Client: %w", err)
+	}
+
+	return &couchDBClient{
+		client: client,
+		cfg:    cfg,
+		logger: logger,
+	}, nil
+}
+
+// Get issues an authorized Get requests to the specified url.
+func (cc *couchDBClient) Get(path string) ([]byte, error) {
+	req, err := cc.BuildReq(path)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := cc.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err = resp.Body.Close(); err != nil {
+			cc.logger.Warn("failed to close response body", zap.Error(err))
+		}
+	}()
+
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
As of go 1.17 the `ReadAll` function has been moved to the `io` package so this should be [io.ReadAll](https://pkg.go.dev/io#ReadAll) instead.

Reply:
updated